### PR TITLE
Improvements to caching

### PR DESCRIPTION
**Pull request by iHiD**
I've added some instructions and tweaked the docker-compose.yml slightly to match the other Dockerfiles.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+This is probably most easily installed via [ruby-install](https://github.com/postmodern/ruby-install):
+```
+ruby-install ruby 2.6
+```
+
+Other Ruby installation options are avaliable on the [ruby-lang website](https://www.ruby-lang.org/en/documentation/installation/).
+
+#### JavaScript
+
+You probably also need some sort of JavaScript/Node but I have no idea how any of that works - so someone might want to update this PR with instructions... :)
```
Comment (iHiD) [inline README.md:76]: We can leave this to one of the JS pps to help with.

File: docker-compose.yml [hunk 2]
```diff
@@ -28,45 +28,36 @@ services:
...
     build:
       context: ../tooling-invoker
       dockerfile: Dockerfile.dev
-    depends_on:
-      - tooling-orchestrator
     environment:
       - EXERCISM_ENV=development
       - EXERCISM_DOCKER=true
-      - ORCHESTRATOR_HOST=tooling-orchestrator
+      - EXERCISM_INVOKE_VIA_DOCKER=true
```
Comment (iHiD) [inline docker-compose.yml:60]: I've added a specific env var for this. This is so we can run in docker but just with the compiled ruby-test-runner without needing the extra dependency of the servers, just to make integration testing a little easier. We'll leave it on by default though.

File: docker-compose.yml [hunk 3]
```diff
@@ -28,45 +28,36 @@ services:
...
       - 3021:3021
-    volumes:
-      - ../tooling-orchestrator:/usr/src/app
-    depends_on:
-      - website
     environment:
       - EXERCISM_ENV=development
       - EXERCISM_DOCKER=true
+    volumes:
+      - ../tooling-orchestrator:/usr/src/app
```
Comment (iHiD) [inline docker-compose.yml:51]: I just reordered this to bring it in line with the others.

File: docker-compose.yml [hunk 4]
```diff
@@ -28,45 +28,36 @@ services:
...
 
   tooling-orchestrator:
     build:
       context: ../tooling-orchestrator
       dockerfile: Dockerfile.dev
     ports:
       - 3021:3021
-    volumes:
-      - ../tooling-orchestrator:/usr/src/app
-    depends_on:
```
Comment (iHiD) [inline docker-compose.yml:54]: We don't need any depends_on, and it makes it slower if we have them, as it waits for one to start before starting the next. Better just to get everything running asap I think.

File: docker-compose.yml [hunk 5]
```diff
@@ -28,45 +28,36 @@ services:
...
     ports:
       - 3020:3020
       - 3035:3035
       - 3334:3334
     volumes:
       - ../v3-website:/usr/src/app
-      - /usr/src/app/node_modules
       - type: tmpfs
-        target: /usr/src/app/tmp
+        target: /usr/src/tmp
```
Comment (iHiD) [inline docker-compose.yml:39]: This might be a bad move. Did you deliberately override the tmp dir in app?

File: docker-compose.yml [hunk 6]
```diff
@@ -28,18 +28,13 @@ services:
     environment:
       - EXERCISM_ENV=development
       - EXERCISM_DOCKER=true
-      - MYSQL_HOST=mysql
-      - MYSQL_PORT=3306
-      - MYSQL_USERNAME=root
-      - MYSQL_PASSWORD=password
```
Comment (iHiD) [inline docker-compose.yml:34]: These are all unused now.

Comment (ErikSchierboom) [reply docker-compose.yml:34]: 👍

File: docker-compose.yml (hunk 4 above)
Comment (ErikSchierboom) [reply docker-compose.yml:54]: Yep, makes sense

File: README.md (hunk 1 above)
Comment (ErikSchierboom) [reply README.md:76]: :)

**Pull request merged**